File: backend/schemas/attachment.py
"""
OrderHub CRM — Attachment Schemas
"""

import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict

from models.attachment import AttachmentType


class AttachmentResponse(BaseModel):
    id: uuid.UUID
    order_id: uuid.UUID
    uploaded_by_id: uuid.UUID
    file_name: str
    original_name: Optional[str] = None
    file_size: int
    mime_type: str
    attachment_type: AttachmentType
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

File: frontend/src/types/attachment.ts
export type AttachmentType = 'mockup' | 'reference' | 'other';

export interface AttachmentResponse {
  id: string;
  order_id: string;
  uploaded_by_id: string;
  file_name: string;
  original_name?: string;
  file_size: number;
  mime_type: string;
  attachment_type: AttachmentType;
  created_at: string;
}
